examples/2d/rotation.rs

```diff
@@ -0,0 +1,253 @@
+    asset_server: Res<AssetServer>,
+) {
+    // Add the game's entities to our world
+
+    let ship_handle = asset_server.load("textures/simplespace/ship_C.png");
+    let enemy_a_handle = asset_server.load("textures/simplespace/enemy_A.png");
+    let enemy_b_handle = asset_server.load("textures/simplespace/enemy_B.png");
+
+    // cameras
+    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
+
+    let horizontal_margin = BOUNDS.x / 4.0;
+    let vertical_margin = BOUNDS.y / 4.0;
+
+    // ship
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(ship_handle.into()),
+            ..Default::default()
+        })
+        .insert(Player {
+            movement_speed: 500.0,                  // metres per second
+            rotation_speed: f32::to_radians(360.0), // degrees per second
+        });
+
+    // snap to player enemy spawns on the bottom and left
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(enemy_a_handle.clone().into()),
+            transform: Transform::from_xyz(0.0 - horizontal_margin, 0.0, 0.0),
+            ..Default::default()
+        })
+        .insert(SnapToPlayer);
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(enemy_a_handle.into()),
+            transform: Transform::from_xyz(0.0, 0.0 - vertical_margin, 0.0),
+            ..Default::default()
+        })
+        .insert(SnapToPlayer);
+
+    // rotate to player enemy spawns on the top and right
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(enemy_b_handle.clone().into()),
+            transform: Transform::from_xyz(0.0 + horizontal_margin, 0.0, 0.0),
+            ..Default::default()
+        })
+        .insert(RotateToPlayer {
+            rotation_speed: f32::to_radians(45.0), // degrees per second
+        });
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(enemy_b_handle.into()),
+            transform: Transform::from_xyz(0.0, 0.0 + vertical_margin, 0.0),
+            ..Default::default()
+        })
+        .insert(RotateToPlayer {
+            rotation_speed: f32::to_radians(90.0), // degrees per second
+        });
+}
+
+/// Demonstrates applying rotation and movement based on keyboard input.
+fn player_movement_system(
+    keyboard_input: Res<Input<KeyCode>>,
+    mut query: Query<(&Player, &mut Transform)>,
+) {
+    let (ship, mut transform) = query.single_mut();
+
+    let mut rotation_factor = 0.0;
+    let mut movement_factor = 0.0;
+
+    if keyboard_input.pressed(KeyCode::Left) {
+        rotation_factor += 1.0;
+    }
+
+    if keyboard_input.pressed(KeyCode::Right) {
+        rotation_factor -= 1.0;
+    }
+
+    if keyboard_input.pressed(KeyCode::Up) {
+        movement_factor += 1.0;
+    }
+
+    // create the change in rotation around the Z axis (pointing through the 2d plane of the screen)
```

Comment:
This _seems_ to imply the Z axis points away from camera (forward), whereas a comment below in the other system says it actually points toward the camera (backward). Can you clarify the comment please?

Reply:
Attempted to clarify. I wasn't intending to talk about the sign of the axis here or right hand rotation, this is covered in a later comment. Perhaps it should be covered here as well, I dunno. This example is pretty simple though so tempted to leave discussions about handedness for the more advanced example further down.